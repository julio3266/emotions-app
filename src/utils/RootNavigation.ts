import {createRef} from 'react';

export const navigationRef: any = createRef();

export const navigate = (name: string, params?: any) => {
  navigationRef.current?.navigate(name, params);
};

export const goBack = () => {
  navigationRef.current?.goBack();
};

export const params = () => {
  navigationRef.current?.params;
};
