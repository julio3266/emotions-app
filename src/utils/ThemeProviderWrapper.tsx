import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import Theme from '../theme/styles';

interface IProps {
  children: any;
  padding?: boolean;
}

export const ThemeProviderWrapper: React.FC<IProps> = ({children}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};
