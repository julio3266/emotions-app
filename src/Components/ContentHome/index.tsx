import React from 'react';
import {navigate} from '@src/utils/RootNavigation';
import {assignTestId} from '@utils/qualityAssurance';
import {BtnEmotion} from '@Components/BtnEmotion';
import * as Styled from './styles';

export interface IContentHome {
  testID?: string;
  emotions?: string[];
}

export const ContentHome: React.FC<IContentHome> = ({
  testID = 'ContentHome',
  emotions,
}) => {
  return (
    <Styled.Content {...assignTestId('View', testID)}>
      <Styled.Column {...assignTestId('View', `${testID}_column`)}>
        {emotions?.map((columnItem: any, columnIndex: any) => {
          return (
            <Styled.Row
              {...assignTestId('View', `${testID}_row`)}
              key={columnIndex.toString()}>
              {columnItem.map((emotionItem: any, emotionIndex: any) => {
                return (
                  <BtnEmotion
                    key={emotionItem.id.toString()}
                    {...assignTestId('View', `${testID}_btn_${emotionIndex}`)}
                    emoji={`${emotionItem.emoji}`}
                    title={`${emotionItem.name}`}
                    handlePress={() =>
                      navigate('Details', {
                        emoji: emotionItem?.emoji,
                        currentEmotion: emotionItem?.name,
                      })
                    }
                  />
                );
              })}
            </Styled.Row>
          );
        })}
      </Styled.Column>
    </Styled.Content>
  );
};
