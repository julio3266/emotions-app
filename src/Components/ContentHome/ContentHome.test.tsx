import React from 'react';
import {render} from '@testing-library/react-native';
import {mockTestID} from '../../utils/qualityAssurance';
import {ThemeProviderWrapper as wrapper} from '../../utils/ThemeProviderWrapper';
import {ContentHome, IContentHome} from '../ContentHome';

describe('Components/ContentHome', () => {
  let props: IContentHome = {
    testID: 'Home',
  };
  const {findByTestId, toJSON} = render(<ContentHome {...props} />, {
    wrapper,
  });

  const sut_container = mockTestID('View', props.testID!).testID;
  const sut_column = mockTestID('View', `${props.testID!}_column`).testID;
  const sut_row = mockTestID('View', `${props.testID!}_row`).testID;
  const sut_btn = mockTestID('View', `${props.testID!}_btn`).testID;

  it(`should render #${sut_container}`, () => {
    const sut = findByTestId(sut_container);
    expect(sut).toBeTruthy();
  });

  it(`should render #${sut_column}`, () => {
    const sut = findByTestId(sut_column);
    expect(sut).toBeTruthy();
  });

  it(`should render #${sut_row}`, () => {
    const sut = findByTestId(sut_row);
    expect(sut).toBeTruthy();
  });

  it(`should render #${sut_btn}`, () => {
    const sut = findByTestId(sut_btn);
    expect(sut).toBeTruthy();
  });

  it(`should render a snapshot`, () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
