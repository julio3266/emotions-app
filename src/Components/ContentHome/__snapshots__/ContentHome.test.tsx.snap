// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/ContentHome should render a snapshot 1`] = `null`;
